src/passes/pass.cpp

```diff
@@ -138,19 +138,22 @@ void PassRunner::addDefaultGlobalOptimizationPasses() {
 }
 
 void PassRunner::run() {
-  if (options.debug) {
+  static int passDebug = getenv("BINARYEN_PASS_DEBUG") ? atoi(getenv("BINARYEN_PASS_DEBUG")) : 0;
+  if (options.debug || passDebug) {
+    // don't recurse pass debug into sub-passes, as it doesn't help anyhow and also is bad for e.g. printing which is a pass
```

Comment:
This comment helped explain why there are two variables for debug, why we set it to 0, how this is expected to work etc.